fixed #10 by trying url patterns

farseerfc opened the pull request:
To solve #10 we will try 3 url patterns now. We may need to re-order the templates list after migration to aur4 is done.

lilydjwg commented:
There's no need to HEAD first. Just get its content and check if it's 200. Then write the data to tar (or maybe use `tarfile`?).

farseerfc commented:
Changed to write data through tarfile. This will save some external command call and file operations.
I added a AurDownloadException which will be raised instead of CalledProcessException from `tar` command, but I am not sure where to catch this exception. 
I added handling for it in lilac, is it enough?

lilydjwg commented on lilaclib.py:
```diff
@@ -111,20 +117,41 @@ def download_official_pkgbuild(name):
+    if response.status_code == 200:
+      logger.debug("downloaded aur tarball '%s' from url '%s'" % (name, url))
+      return response.content
+  logger.error("failed to find aur url for '%s'" % name)
```
你不需要自己格式化字符串，这个工作是 logging 内部会在需要的时候做的。

lilydjwg commented on lilaclib.py:
```diff
@@ -111,20 +117,41 @@ def download_official_pkgbuild(name):
+        continue
+      if remain in ('.AURINFO', '.SRCINFO'):
+        continue
+      contentfile = open(remain, "wb")
```
use `with`.

lilydjwg commented on lilaclib.py:
```diff
@@ -111,20 +117,41 @@ def download_official_pkgbuild(name):
+    if tar_file:
+      tar_file.close()
+    if tar_content:
+      tar_content.close()
```
No need to close a `BytesIO`. Also tarfiles support `with` so you don't need to try.
